diff --git core/src/Cuda/Kokkos_Cuda_Instance.cpp core/src/Cuda/Kokkos_Cuda_Instance.cpp
index d57e197b5..e535be90f 100644
--- core/src/Cuda/Kokkos_Cuda_Instance.cpp
+++ core/src/Cuda/Kokkos_Cuda_Instance.cpp
@@ -87,6 +87,7 @@ __global__ void query_cuda_kernel_arch(int *d_arch) {
   *d_arch = 0;
 #endif
 #endif
+  printf("GPU printf: %i\n", *d_arch);
 }
 
 /** Query what compute capability is actually launched to the device: */
@@ -105,6 +106,9 @@ int cuda_kernel_arch(int device_id) {
 
   KOKKOS_IMPL_CUDA_SAFE_CALL(
       cudaMemcpy(&arch, d_arch, sizeof(int), cudaMemcpyDefault));
+
+  KOKKOS_IMPL_CUDA_SAFE_CALL(cudaDeviceSynchronize());
+  printf("CPU printf: %i\n", arch);
   KOKKOS_IMPL_CUDA_SAFE_CALL(cudaFree(d_arch));
   return arch;
 }
