--- core/unit_test/TestTypeInfo.cpp
+++ core/unit_test/TestTypeInfo.cpp
@@ -25,7 +25,7 @@ using Lambda = decltype(func);
 #if defined(__NVCC__) && !defined(__CUDA_ARCH__)
 // can't do much
 // it looks like that there is 1st an EDG pass and then a host pass and they cannot both agree on what the type info is
-#elif defined(__EDG__) || (defined(__NVCC__) && defined(__CUDA_ARCH__))
+#elif defined(__EDG__) || (defined(__NVCC__) && !defined(__clang__) && defined(__CUDA_ARCH__))
 static_assert(TypeInfo<Foo>::name()      == "<unnamed>::Foo");
 static_assert(TypeInfo<FooAlias>::name() == "<unnamed>::Foo");
 static_assert(TypeInfo<Bar>::name()      == "<unnamed>::Bar");
