diff --git a/src/gromacs/mdrun/runner.cpp b/src/gromacs/mdrun/runner.cpp
index f6d5491..1e9c598 100644
--- a/src/gromacs/mdrun/runner.cpp
+++ b/src/gromacs/mdrun/runner.cpp
@@ -2486,7 +2486,7 @@ int Mdrunner::mdrunner()

         // Only release GPU when not using torch, as torch keeps some device handles that
         // only get cleaned up on process exit
-        if (GMX_GPU && !GMX_TORCH)
+        if (GMX_GPU && !GMX_TORCH && !GMX_THREAD_MPI)
         {
             const bool haveDetectedOrForcedCudaAwareMpi =
                     (gmx::checkMpiCudaAwareSupport() == gmx::GpuAwareMpiStatus::Supported
