--- include/cub/util_debug.cuh
+++ include/cub/util_debug.cuh
@@ -300,12 +300,9 @@
 #endif
 }
 #ifndef __CUDA_ARCH__
-#define _CubLog(format, ...) CUB_NS_QUALIFIER::va_printf(format, __VA_ARGS__);
+#define _CubLog(format, ...) printf(format, __VA_ARGS__);
 #else
-#define _CubLog(format, ...)                                                   \
-  CUB_NS_QUALIFIER::va_printf("[block (%d,%d,%d), thread "                     \
-                              "(%d,%d,%d)]: " format,                          \
-                              __VA_ARGS__);
+#define _CubLog(format, ...) printf("[block (%d,%d,%d), thread (%d,%d,%d)]: " format, __VA_ARGS__);
 #endif
 #endif
 #endif
