--- include/thrust/system/cuda/detail/uninitialized_fill.h
+++ include/thrust/system/cuda/detail/uninitialized_fill.h
@@ -57,13 +57,7 @@
     void THRUST_DEVICE_FUNCTION operator()(Size idx)
     {
       value_type& out = raw_reference_cast(items[idx]);
-
-#if defined(__CUDA__) && defined(__clang__)
-      // XXX unsafe. cuda-clang is seemingly unable to call ::new in device code
-      out = value;
-#else
       ::new (static_cast<void *>(&out)) value_type(value);
-#endif
     }
   };    // struct functor
