diff --git a/samples/mlp_learning_an_image.cu b/samples/mlp_learning_an_image.cu
index 962e4b1..18be437 100644
--- a/samples/mlp_learning_an_image.cu
+++ b/samples/mlp_learning_an_image.cu
@@ -136,8 +136,7 @@ int main(int argc, char* argv[]) {
 				{"n_bins", 32},
 			}},
 			{"network", {
-				{"otype", "FullyFusedMLP"},
-				// {"otype", "CutlassMLP"},
+				{"otype", "CutlassMLP"},
 				{"n_neurons", 64},
 				{"n_hidden_layers", 4},
 				{"activation", "ReLU"},
@@ -237,7 +236,7 @@ int main(int argc, char* argv[]) {
 		std::shared_ptr<Loss<precision_t>> loss{create_loss<precision_t>(loss_opts)};
 		std::shared_ptr<Optimizer<precision_t>> optimizer{create_optimizer<precision_t>(optimizer_opts)};
 		std::shared_ptr<NetworkWithInputEncoding<precision_t>> network = std::make_shared<NetworkWithInputEncoding<precision_t>>(n_input_dims, n_output_dims, encoding_opts, network_opts);
-		network->set_jit_fusion(tcnn::supports_jit_fusion());
+		network->set_jit_fusion(false);
 
 		if (network->jit_fusion()) {
 			std::cout << "JIT fusion is enabled." << std::endl;
