--- extern/cmake_utils/FindThrust.cmake
+++ extern/cmake_utils/FindThrust.cmake
@@ -53,9 +53,9 @@ if( THRUST_INCLUDE_DIR )
     )

-  string( REGEX MATCH "^[0-9]" major ${version} )
+  string( REGEX MATCH "^[0-9]" major "${version}" )
-  string( REGEX REPLACE "^${major}00" "" version "${version}" )
+  string( REGEX REPLACE "^-${major}00" "" version "-${version}" )
-  string( REGEX MATCH "^[0-9]" minor ${version} )
+  string( REGEX MATCH "^[0-9]" minor "${version}" )
-  string( REGEX REPLACE "^${minor}0" "" version "${version}" )
+  string( REGEX REPLACE "^-${minor}0" "" version "-${version}" )
   set( THRUST_VERSION "${major}.${minor}.${version}")
   set( THRUST_MAJOR_VERSION "${major}")
   set( THRUST_MINOR_VERSION "${minor}")
