diff --git a/ggml/src/ggml-cuda/common.cuh b/ggml/src/ggml-cuda/common.cuh
--- a/ggml/src/ggml-cuda/common.cuh
+++ b/ggml/src/ggml-cuda/common.cuh
@@ -372,11 +372,11 @@
 }

 static constexpr __device__ int ggml_cuda_get_physical_warp_size() {
-#if defined(GGML_USE_HIP) && (defined(__GFX9__) || defined(__GFX8__))
-    return 64;
+#if defined(__AMDGCN_WAVEFRONT_SIZE__)
+    return __AMDGCN_WAVEFRONT_SIZE__;
 #else
     return 32;
-#endif // defined(GGML_USE_HIP) && (defined(__GFX9__) || defined(__GFX8__))
+#endif // defined(__AMDGCN_WAVEFRONT_SIZE__)
 }

 // Maximum number of bytes that can be copied in a single instruction.
