diff --git a/torch/headeronly/macros/Macros.h b/torch/headeronly/macros/Macros.h
index 558edb1..3d49eda 100644
--- a/torch/headeronly/macros/Macros.h
+++ b/torch/headeronly/macros/Macros.h
@@ -431,7 +431,11 @@ __host__ __device__
     // exactly how it is in glibc in case parts of the program are compiled with
     // different NDEBUG settings. Otherwise we might get 'ambiguous declaration'
     // error. Note: On ROCm - this declaration serves for host side compilation.
+#if defined(__CUDACC__) || defined(__CUDA__)
+    __host__ __device__ void
+#else
     void
+#endif
     __assert_fail(
         const char* assertion,
         const char* file,
